lib/jekyll/readers/theme_assets_reader.rb

```diff
@@ -0,0 +1,47 @@
+module Jekyll
+  class ThemeAssetsReader
+    attr_reader :site
+    def initialize(site)
+      @site = site
+    end
+
+    def read
+      return unless site.theme && site.theme.assets_path
+
+      Find.find(site.theme.assets_path) do |path|
+        next if File.directory?(path)
+        if File.symlink?(path)
+          Jekyll.logger.warn "Theme reader:", "Ignored symlinked asset: #{path}"
+        else
+          read_theme_asset(path)
+        end
+      end
+    end
+
+    private
+    def read_theme_asset(path)
+      base = site.theme.root
+      dir = File.dirname(path.sub("#{site.theme.root}/", ""))
```

Comment:
Would be nice to have a "relative_from" method or something but it depends on what you need. In this case, I want all directories within the theme directory without the forward slash at the beginning (e.g. `assets`). If you do `path.sub(site.theme.root, "") `, you get `/assets` and we can't use Pathutil because it doesn't work on Windows the way we're using it.